### PR TITLE
Not overwrite plugin object if it exists

### PR DESCRIPTION
**Pull request by mshamota-nr**

### CODE REVIEW

File: record/record.go [hunk 1]
```diff
@@ -41,10 +42,13 @@ func RemapRecord(inputRecord FluentBitRecord, inputTimestamp interface{}, plugin
 	if !ok {
 		source = "BARE-METAL"
 	}
-	outputRecord["plugin"] = map[string]string{
-		"type":    "fluent-bit",
-		"version": pluginVersion,
-		"source":  source,
+	_, ok = outputRecord["plugin"]
+	if !ok {
```
Comment (jsubirat) [inline record/record.go:46]: Cool!! You can use this shorter go-ish notation form:
```
if _, ok = outputRecord["plugin"]; !ok {
```

File: record/record_test.go [hunk 2]
```diff
@@ -48,6 +48,22 @@ var _ = Describe("Out New Relic", func() {
...
+		It("Doesn't rewrite plugin.type if it exits", func() {
+			inputMap := make(FluentBitRecord)
+			var inputTimestamp interface{}
+			inputTimestamp = output.FLBTime{
+				time.Now(),
+			}
+			expectedType := "something"
+			pluginMapIn := make(map[string]string)
+			pluginMapIn["type"] = expectedType
+			inputMap["plugin"] = pluginMapIn
```
Comment (jsubirat) [inline record/record_test.go:61]: You can also write this in this compacted way:
```
inputMap["plugin"] := map[string]string{
    "type": expectedType,
}
```

I'd also suggest to rename `inputMap` into `inputRecord`, and `foundOutput` into `remappedRecord`.

**Review by jsubirat — COMMENTED** [review]
Thanks for the work, @mshamota-nr ! 👏

File: record/record_test.go (hunk 2 above)
Comment (jsubirat) [reply record/record_test.go:61]: I just realize you used the same naming as in the other tests. Take this last suggestion as a pure recommendation, then. Feel free to also rename them in the rest of the tests.

**Review by jodstrcil — APPROVED** [review]

File: record/record_test.go (hunk 2 above)
Comment (mshamota-nr) [reply record/record_test.go]: Done!

File: record/record.go (hunk 1 above)
Comment (mshamota-nr) [reply record/record.go]: Done!

File: record/record_test.go [hunk 3]
```diff
@@ -102,7 +102,7 @@ var _ = Describe("Out New Relic", func() {
 
 		It("Correctly handles a JSON array in a []interface{}", func() {
 			// Given
-			inputRecord := map[interface{}]interface{}{
+			inputMap := map[interface{}]interface{}{
```
Comment (jsubirat) [inline record/record_test.go:105]: I actually meant the opposite: rename inputMap to inputRecord everywhere 🤣 But that's OK, we can merge this code and do it later.

**Review by jsubirat — APPROVED** [review]

**Comment by mshamota-nr** [comment]
Jenkins!

**Review by jodstrcil — APPROVED** [review]

**Comment by jodstrcil** [comment]
ok jenkins

**Review by jsubirat — APPROVED** [review]

**Pull request merged**